plugins/qsi-dock/fu-qsi-dock-mcu-device.c

```diff
@@ -0,0 +1,703 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2022 Kevin Chen <hsinfu.chen@qsitw.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-qsi-dock-child-device.h"
+#include "fu-qsi-dock-common.h"
+#include "fu-qsi-dock-mcu-device.h"
+
+struct _FuQsiDockMcuDevice {
+	FuHidDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuQsiDockMcuDevice, fu_qsi_dock_mcu_device, FU_TYPE_HID_DEVICE)
+
+#define FU_QSI_DOCK_MCU_DEVICE_TIMEOUT 90000 /* ms */
+
+#define FU_QSI_DOCK_DEVICE_FLAG_VERFMT_HP (1 << 0)
+
+static gboolean
+fu_qsi_dock_mcu_device_tx(FuQsiDockMcuDevice *self,
+			  guint8 CmdPrimary,
+			  guint8 CmdSecond,
+			  const guint8 *inbuf,
+			  gsize inbufsz,
+			  GError **error)
+{
+ 	guint8 buf[64] = {0};
+	buf[0] = Report_ID;
+	buf[1] = CmdPrimary;
+	buf[2] = CmdSecond;
```

Comment:
you can do this using:

    guint8 buf[64] = { Report_ID, CmdPrimary, CmdSecond };